internal/controlplane/handlers_authz.go

```diff
@@ -567,29 +572,22 @@ func (s *Server) removeRole(
 	subject string,
 ) (*minder.RemoveRoleResponse, error) {
 	var err error
-	// We may be given a human-readable identifier which can vary over time. Resolve
-	// it to an IDP-specific stable identifier so that we can support subject renames.
-	identity := &auth.Identity{
-		Provider:  nil,
-		UserID:    subject,
-		HumanName: subject,
-	}
-	if flags.Bool(ctx, s.featureFlags, flags.IDPResolver) {
-		identity, err = s.idClient.Resolve(ctx, subject)
-		if err != nil {
-			zerolog.Ctx(ctx).Error().Err(err).Msg("error resolving identity")
-			return nil, util.UserVisibleError(codes.NotFound, "could not find identity %q", subject)
-		}
+	// Resolve the subject to an identity
+	identity, err := s.idClient.Resolve(ctx, subject)
+	if err != nil {
+		zerolog.Ctx(ctx).Error().Err(err).Msg("error resolving identity")
+		return nil, util.UserVisibleError(codes.NotFound, "could not find identity %q", subject)
 	}
 
 	// Verify if user exists
-	if _, err := s.store.GetUserBySubject(ctx, identity.UserID); err != nil {
+	if _, err := s.store.GetUserBySubject(ctx, identity.String()); err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
 			return nil, util.UserVisibleError(codes.NotFound, "User not found")
 		}
 		return nil, status.Errorf(codes.Internal, "error getting user: %v", err)
 	}
```

Comment:
Why do we need this at all?

Reply:
This was there prior to these changes, but my understanding is to fail early if we try to do something on a user it doesn't exist.